### PR TITLE
feat: move generated ':root' element to themeLayer's first

### PR DESCRIPTION
**Pull request by GavinXue**
With last version, When we put `class="dark"`(or other themes) in html root element, It will not change the theme, because the dark theme will covered by `:root` vars. Just like under image display.  

move root first img

Maybe moving the `:root` element to first make more sense (also changed the test case). if there is any problem I didn't care about, please let me know.

### CODE REVIEW

**Comment by Dunqing** [comment]
Hello, Thank you for your contribution.

The `selector` option is able to change the `:root` to anything, so this will cause the sort function not to work, so do we need to have the option to re-order all themes in order, e.g. a `themeSort` option?

The option looks like this

```ts
{
  themeSort: {
     light: 0,
     dark: 1,
     other: 2
  }
}
```

Do you think this is ok?

**Comment by GavinXue** [comment]
Hi，in my opinion,Add a theme sort maybe make the preset a bit more complicated. Because two themes will not used at same time in most of circumstances. Except the root.
Someone who want to use multi themes at same time, should manage this themselves by their code sorts.
If we offer this option,developer will care about the settings (if we have many, we should set carefully) and their code class sort. Maybe giving a option at the time someone needed will be better.
anyway, if we want, I can try to make it. 
Waiting for your reply. Thanks😄

**Comment by Dunqing** [comment]
> Hi，in my opinion,Add a theme sort maybe make the preset a bit more complicated. Because two themes will not used at same time in most of circumstances. Except the root. Someone who want to use multi themes at same time, should manage this themselves by their code sorts. If we offer this option,developer will care about the settings (if we have many, we should set carefully) and their code class sort. Maybe giving a option at the time someone needed will be better. anyway, if we want, I can try to make it. Waiting for your reply. Thanks😄

Yes, I agree with you. That's what I'm worried about. It'll make it complicated.

File: src/index.ts [hunk 1]
```diff
@@ -167,6 +167,10 @@ export function presetTheme<T extends Record<string, any>>(options: PresetThemeO
               line = line.replace(/\..*?\s(.*\{)/, '$1')
             }
             return line
+          }).sort((a, b) => {
+            if (a.match(/^:root|^@media|^\}/)?.length)
```
Comment (Dunqing) [inline src/index.ts:171]: you need to get the light's selector and use it to sort the theme.

File: src/index.ts [hunk 2]
```diff
@@ -167,6 +167,10 @@ export function presetTheme<T extends Record<string, any>>(options: PresetThemeO
...
+              return b.match(/^:root|^@media|^\}/)?.length ? 0 : -1
```
Comment (Dunqing) [inline src/index.ts:172]: Here two lines of code use the same RegExp, So we should store the RegExp to a new variable and then use it

**Review by Dunqing — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (GavinXue) [reply src/index.ts:171]: Yes, that's a problem I didn't catch, I'll fix it.

File: src/index.ts (hunk 2 above)
Comment (GavinXue) [reply src/index.ts:172]: Ok, I'll make it clean.

Comment (GavinXue) [reply src/index.ts]: Fix it, please have a check.

**Review by Dunqing — APPROVED** [review]

**Comment by Dunqing** [comment]
Thank you for your great work, I will merge it and publish a new version later

**Pull request merged**